versioned_docs/version-7.x/testing.md

```diff
@@ -54,59 +48,855 @@ Then we need to use this setup file in our jest config. You can add it under `se
 
 ## Writing tests
 
 We recommend using [React Native Testing Library](https://callstack.github.io/react-native-testing-library/) along with [`jest-native`](https://github.com/testing-library/jest-native) to write your tests.
 
-Example:
+We will go through some real-world case test code examples. Each code example consists of tested navigator and test code file.
+
+### Example 1
+
+Navigate to settings screen by "Go to Settings" button press.
+
+<Tabs groupId="example" queryString="example">
+<TabItem value="static" label="Static" default>
+
+```js
+import { useNavigation } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { Button, Text, View } from 'react-native';
+
+const HomeScreen = () => {
+  const navigation = useNavigation();
+
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Home screen</Text>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Go to Settings"
+      />
+    </View>
+  );
+};
+
+const SettingsScreen = () => {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Settings screen</Text>
+    </View>
+  );
+};
+
+export const StackNavigator = createStackNavigator({
+  screens: {
+    Home: HomeScreen,
+    Settings: SettingsScreen,
+  },
+});
+```
+
+</TabItem>
+<TabItem value="dynamic" label="Dynamic">
+
+```js
+import { createStackNavigator } from '@react-navigation/stack';
+import { Button, Text, View } from 'react-native';
+
+const HomeScreen = ({ navigation }) => {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Home screen</Text>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Go to Settings"
+      />
+    </View>
+  );
+};
+
+const SettingsScreen = () => {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Settings screen</Text>
+    </View>
+  );
+};
+
+export const StackNavigator = () => {
+  const Stack = createStackNavigator();
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={HomeScreen} />
+      <Stack.Screen name="Settings" component={SettingsScreen} />
+    </Stack.Navigator>
+  );
+};
+```
+
+</TabItem>
+</Tabs>
+
+<Tabs groupId="example" queryString="example">
+<TabItem value="static" label="Static" default>
+
+```js
+import { expect, test } from '@jest/globals';
+import { createStaticNavigation } from '@react-navigation/native';
+import { fireEvent, render, screen } from '@testing-library/react-native';
+
+import { StackNavigator } from './StackNavigator';
+
+test('navigates to settings by "Go to Settings" button press', () => {
+  const StackNavigation = createStaticNavigation(StackNavigator);
+  render(<StackNavigation />);
+
+  fireEvent.press(screen.queryByText('Go to Settings'));
```

Comment:
Could you use `getBy…` instead of `queryBy…` when we expect that an element will be present on the screen, and use `queryBy…` only when we expect that an element won’t be available?
- This way, it will be clearer which elements we should expect to be visible or not.
- In case of an error, `getBy…` will fail earlier than `queryBy….`